idaes/models/properties/modular_properties/base/generic_property.py

```diff
@@ -653,6 +706,149 @@ def _bubble_dew_scaling(
+                    overwrite=overwrite,
+                )
+                self.scale_constraint_by_component(
+                    model.true_mole_frac_constraint[p, j],
+                    model.flow_mol_phase_comp_true[p, j],
+                    overwrite=overwrite,
+                )
+            else:
+                self.scale_constraint_by_component(
+                    model.appr_to_true_species[p, j],
+                    model.flow_mol_phase_comp_true[p, j],
+                    overwrite=overwrite,
+                )
+                self.scale_constraint_by_component(
+                    model.true_mole_frac_constraint[p, j],
+                    model.flow_mol_phase_comp_true[p, j],
+                    overwrite=overwrite,
+                )
+
+    def _true_basis_variable_scaling_routine(self, model, overwrite):
+        """
+        Method to scale variables created when using a true species
+        basis for systems with inherent reactions.
+
+        Args:
+            model: GenericStateBlockData which is being scaled
+            overwrite: Boolean flag about whether or not to
+                overwrite existing scaling factors
+        Returns:
+            None
+        """
+        sf_mole_frac_pc_apparent = {}
+        for idx, vardata in model.mole_frac_phase_comp_apparent.items():
+            self.scale_variable_by_default(vardata, overwrite=overwrite)
+            sf_mole_frac_pc_apparent[idx] = self.get_scaling_factor(vardata)
+
+        sf_flow_mol_phase = {}
+        for p in model.phase_list:
+            sf_flow_mol_phase[p] = self.get_scaling_factor(model.flow_mol_phase[p])
+
+        for (p, j), vardata in model.flow_mol_phase_comp_apparent.items():
+            sf = sf_flow_mol_phase[p] * sf_mole_frac_pc_apparent[(p, j)]
+            self.set_variable_scaling_factor(vardata, sf, overwrite=overwrite)
+
+    def _true_basis_constraint_scaling_routine(self, model, overwrite):
+        """
+        Method to scale variables created when using a true species
+        basis for systems with inherent reactions.
+
+        Args:
+            model: GenericStateBlockData which is being scaled
+            overwrite: Boolean flag about whether or not to
+                overwrite existing scaling factors
+        Returns:
+            None
+        """
+        # Autoscaling for constraints from a true species basis
+        # was not implemented in the legacy scaling tools.
+        # Use default scaling until we can rework the true to apparent
+        # species conversion
+        for condata in model.true_to_appr_species.values():
+            self.scale_constraint_by_nominal_value(condata, overwrite=overwrite)
+        for condata in model.appr_mole_frac_constraint.values():
+            self.scale_constraint_by_nominal_value(condata, overwrite=overwrite)
+
 
 # TODO: Set a default state definition
 # TODO: Need way to dynamically determine units of measurement....
```

Comment:
Can the UOM be determined from the constraint itself, or using flow basis and/or stoichiometry?